internal/conns/config.go

```diff
@@ -191,9 +193,25 @@ func (c *Config) ConfigureProvider(ctx context.Context, client *AWSClient) (*AWS
 		}
 	}
 
+	// Fetch tag policy details when enforced
+	if c.TagPolicyConfig != nil {
+		tflog.Debug(ctx, "Retrieving tag policy details")
+		reqTags, err := tagpolicy.GetRequiredTags(ctx, cfg)
+		if err != nil {
```

Comment:
This is scary to me because rate limit, unavailable API, can't put my finger on it concerns about some poor sysop locked out, config is locked down, etc.

I get that compliance is vital but what if we waited to error until we actually did a verification? Not sure that's possible but if they don't create resource / update tags, they could still do things like empty plan which wouldn't require required tags even when tags are acting up.

We could also lazy load required tags when/if they are needed.

Reply:
Yes, wouldn't want to see all operations paused while the AWS SDK does a 20 minute throttling retry...

Reply:
I will open a follow-up issue to refactor the `ListRequiredTags` API call timing.